### DISCUSSION TITLE
How to access Global Environment Variable From Workflow

### DISCUSSION THREAD

**Discussion by RizanPSTU** [codemagic-ci-cd/codemagic-docs#2168] [Q&A]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

Tried to access from the sh file

<img width="724" alt="Screenshot 2023-11-03 at 3 56 45 PM" src="https://github.com/codemagic-ci-cd/codemagic-docs/assets/32144390/e0582e6f-6065-465a-90a9-f52ab597a37f">

called the sh file on pre build 

<img width="1198" alt="Screenshot 2023-11-03 at 3 56 38 PM" src="https://github.com/codemagic-ci-cd/codemagic-docs/assets/32144390/e81a3ae1-063f-4b48-8450-5b3bf5cda8d3">

Here are some env variables 

<img width="1496" alt="Screenshot 2023-11-03 at 3 56 27 PM" src="https://github.com/codemagic-ci-cd/codemagic-docs/assets/32144390/026e4f37-3cc1-4f24-bb72-536323afda7a">

### Expected results

Can't access the Global Environment variable from team. How to access the Global team Env values?

### Actual results

What to access team values

**Comment by RizanPSTU** [comment]
Anyone?

**Comment by RizanPSTU** [comment]
@Ycoi anyone?

**Comment by RizanPSTU** [comment]
ANYONE? Can say anything?

**Reply by dtrdic** [reply]
hello @RizanPSTU, you can store your variables in a settings.env file, then download the settings.env file in post clone section, and write the values to the CM_ENV environment variable.
So let's say you have this variable inside your settings.env file
![image](https://github.com/codemagic-ci-cd/codemagic-docs/assets/38918425/4004d7d2-a89c-40c3-a1fd-966d96e6877b)

then in your workflow editor in the post-clone script you will append the content of the TEST_SERVICE_ACCOUNT_CREDENTIALS variable to the CM_ENV environment variable like this;

![image](https://github.com/codemagic-ci-cd/codemagic-docs/assets/38918425/8c881ad8-7ed4-4bd1-bbf9-32003f2f1586)

then call the variable like this;
![image](https://github.com/codemagic-ci-cd/codemagic-docs/assets/38918425/18f36e00-5cd5-467b-b3e0-085fcfa676ee)

Btw catching variables from `Teams> Global variables and secrets` are possible only when using codemagic.yaml configuration

<img width="484" alt="image" src="https://github.com/codemagic-ci-cd/codemagic-docs/assets/38918425/c6dd474d-ad99-45ca-a78d-8f2337c12a9d">


Use the application settings page for storing and catching your environment variables instead;

<img width="1247" alt="Screenshot 2023-11-08 at 15 33 52" src="https://github.com/codemagic-ci-cd/codemagic-docs/assets/38918425/d3caca57-b4dc-4f45-8177-1739854fb7b5">

**Comment by dtrdic** [comment]
I'm tentatively closing the discussion due to a lack of activity. You can open the discussion again by replying to this message or opening a new discussion.

**Reply by AnaNedicGithub** [reply]
I have the same problem now: global variables aren't accessible from codemagic.yaml
